forest/cli/src/cli/snapshot_fetch.rs

```diff
@@ -202,6 +220,66 @@ where
     Ok(())
 }
 
+async fn download_snapshot_and_validate_checksum_with_aria2<C>(
+    client: hyper::Client<C>,
+    url: Url,
+    snapshot_path: &Path,
+) -> anyhow::Result<()>
+where
+    C: Connect + Clone + Send + Sync + 'static,
+{
+    info!("Snapshot url: {url}");
+    info!("Snapshot will be downloaded to {}", snapshot_path.display());
+
+    if let Err(err) = which::which("aria2c") {
+        bail!("Command aria2c is not available. To use --aria2, make sure you have aria2 installed. E.g. sudo apt-get install -y aria2 or brew intall aria2 or follow the instructions on https://aria2.github.io/ \n{err}");
+    }
+
+    let checksum_url = replace_extension_url(url.clone(), "sha256sum")?;
+    let checksum_response = client.get(checksum_url.as_str().try_into()?).await?;
+    if !checksum_response.status().is_success() {
+        bail!("Unable to get the checksum file. Snapshot downloaded but not verified. Url: {checksum_url}");
```

Comment:
This is not precisely true given the current logic, no? The user will get the message `Snapshot downloaded but not verified (...)`, but nothing is there, right?

Reply:
Ah good catch, I copy & paste the text w/o reading it : )

Reply:
Fixed